homeassistant/components/firmata/board.py

```diff
@@ -0,0 +1,140 @@
+"""Code to handle a Firmata board."""
+import logging
+
+from pymata_express.pymata_express import PymataExpress
+
+from homeassistant.const import CONF_NAME
+
+from .const import (
+    CONF_ARDUINO_INSTANCE_ID,
+    CONF_ARDUINO_WAIT,
+    CONF_BINARY_SENSORS,
+    CONF_SAMPLING_INTERVAL,
+    CONF_SERIAL_BAUD_RATE,
+    CONF_SERIAL_PORT,
+    CONF_SLEEP_TUNE,
+    CONF_SWITCHES,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FirmataBoard:
+    """Manages a single Firmata board."""
+
+    def __init__(self, hass, config_entry):
+        """Initialize the board."""
+        self.config_entry = config_entry
+        self.config = self.config_entry.data
+        self.hass = hass
+        self.available = True
+        self.api = None
+        self.firmware_version = None
+        self.protocol_version = None
+        self.name = self.config[CONF_NAME]
+        self.switches = []
+        self.binary_sensors = []
+        self.used_pins = []
+        self.board_info = {
+            "connections": {},
+            "identifiers": {(DOMAIN, self.name)},
+            "manufacturer": "Firmata",
+            "name": self.name,
+        }
+
+    async def async_setup(self, tries=0):
+        """Set up a Firmata instance."""
+        try:
+            _LOGGER.info("Connecting to Firmata %s", self.name)
+            self.api = await get_board(self.config)
+            self.firmware_version = await self.api.get_firmware_version()
+            self.board_info["sw_version"] = self.firmware_version
```

Comment:
Setting the firmware version doesn't need to be within the try... except block. We only want lines that can raise here. Use an `else:` block if needed.